packages/devtools_app/test/test_infra/flutter_test_driver.dart

```diff
@@ -311,28 +311,32 @@ class FlutterRunTestDriver extends FlutterTestDriver {
     FlutterRunConfiguration runConfig = const FlutterRunConfiguration(),
     File? pidFile,
   }) async {
+    // Start listening for all events before we start printing
+    // to stdout. This avoids race conditions and deadlocks.
+    final Future<Map<String, dynamic>> startedFuture =
+        waitFor(event: 'app.started', timeout: appStartTimeout);
+    final connectedFuture = waitFor(event: 'daemon.connected');
+    final debugPortFuture =
+        waitFor(event: 'app.debugPort', timeout: appStartTimeout);
+
     await super.setupProcess(
       args,
       flutterExecutable: flutterExecutable,
       runConfig: runConfig,
       pidFile: pidFile,
     );
 
+    // Await the startedFuture, which signals that the app has started.
+    _currentRunningAppId = (await startedFuture)['params']['appId'];
```

Comment:
do we expect this to happen before the `connected` future? Does it matter what order these complete in? If not, it may be safer to await both at once using `Future.wait`

Reply:
It shouldn't matter what order they are awaited. 
In the  logs we would expect `daemon.connected` to happen earlier than `app.started`.
So waiting on the `app.started` future first just means that we would expect the `daemon.connected` future to be ready to complete immediately upon await.